pkg/reconciler/domainmapping/reconciler.go

```diff
@@ -292,19 +297,36 @@ func (r *Reconciler) resolveRef(ctx context.Context, dm *v1alpha1.DomainMapping)
 
 func (r *Reconciler) reconcileDomainClaim(ctx context.Context, dm *v1alpha1.DomainMapping) error {
 	dc, err := r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Get(ctx, dm.Name, metav1.GetOptions{})
+	if err != nil && !apierrs.IsNotFound(err) {
+		return fmt.Errorf("failed to get ClusterDomainClaim: %w", err)
+	}
+
 	if apierrs.IsNotFound(err) {
-		if dc, err = r.netclient.NetworkingV1alpha1().ClusterDomainClaims().Create(ctx, resources.MakeDomainClaim(dm), metav1.CreateOptions{}); err != nil {
-			return fmt.Errorf("failed to create ClusterDomainClaim: %w", err)
+		dc, err = r.createDomainClaimIfAllowed(ctx, dm)
+		if err != nil {
+			return err
 		}
-	} else if err != nil {
-		return fmt.Errorf("failed to get ClusterDomainClaim: %w", err)
 	}
 
 	if dm.Namespace != dc.Spec.Namespace {
 		dm.Status.MarkDomainClaimNotOwned()
-		return fmt.Errorf("domain mapping: namespace %q does not own cluster domain claim for %q", dm.Namespace, dm.Name)
+		return fmt.Errorf("namespace %q does not own ClusterDomainClaim for %q", dm.Namespace, dm.Name)
 	}
 
 	dm.Status.MarkDomainClaimed()
 	return nil
 }
+
+func (r *Reconciler) createDomainClaimIfAllowed(ctx context.Context, dm *v1alpha1.DomainMapping) (*netv1alpha1.ClusterDomainClaim, error) {
```

Comment:
this name presumes there's a `createDC` without any allowance check. I think just `createDomainClaim` is fine, especially that it's an internal method.

Reply:
done